### PR TITLE
feat: Add permissions to the database-load module to allow deletion of the …

### PR DESCRIPTION
**Pull request by antroy-madetech**
…dump file from S3

### CODE REVIEW

File: postgres/database-load/tests/unit.tftest.hcl [hunk 1]
```diff
@@ -135,53 +135,45 @@ run "data_load_unit_test" {
   }
 
   assert {
-    condition     = length(data.aws_iam_policy_document.data_load.statement) == 2
+    condition     = length(data.aws_iam_policy_document.data_load.statement) == 3
     error_message = "Should be 1 policy statement"
```
Comment (ksugden) [inline postgres/database-load/tests/unit.tftest.hcl:139]:
```suggestion
    error_message = "Should be 3 policy statements"
```

File: postgres/database-load/tests/unit.tftest.hcl [hunk 2]
```diff
@@ -135,53 +135,45 @@ run "data_load_unit_test" {
...
   assert {
-    condition     = contains(data.aws_iam_policy_document.data_load.statement[0].actions, "s3:ListBucket")
-    error_message = "Permission not found: s3:ListBucket"
-  }
-
-  assert {
-    condition     = contains(data.aws_iam_policy_document.data_load.statement[0].actions, "s3:GetObject")
-    error_message = "Permission not found: s3:GetObject"
+    condition     = length(data.aws_iam_policy_document.data_load.statement[0].actions) == 6
+    error_message = "Should be 7 permissions on policy statement"
```
Comment (ksugden) [inline postgres/database-load/tests/unit.tftest.hcl:144]:
```suggestion
    error_message = "Should be 6 permissions on policy statement"
```

File: postgres/database-load/main.tf [hunk 3]
```diff
@@ -65,20 +67,33 @@ resource "aws_iam_role_policy" "allow_task_creation" {
...
 
 data "aws_iam_policy_document" "data_load" {
-  # checkov:skip=CKV_AWS_356:Permissions required to upload
   policy_id = "data_load"
   statement {
     sid    = "AllowReadFromS3"
     effect = "Allow"
 
-    actions = local.s3_permissions
+    actions = concat(local.s3_permissions, ["s3:DeleteObject"])
```
[the remark this answers is not in the document]
Comment (antroy-madetech) [reply postgres/database-load/main.tf:75]: We only want it on the load, not the dump and the local is used by both

File: postgres/tests/unit.tftest.hcl [hunk 4]
```diff
@@ -348,17 +348,97 @@ run "aws_db_instance_unit_test" {
...
+          from = "test-environment"
+          to   = "some-prod-environment"
+        }
+      ]
+    }
+  }
+
+  assert {
+    condition     = length(module.database-dump) == 0
+    error_message = "database-dump module should be not created"
```
Comment (ksugden) [inline postgres/tests/unit.tftest.hcl:435]:
```suggestion
    error_message = "database-dump module should not be created"
```

File: postgres/tests/unit.tftest.hcl [hunk 5]
```diff
@@ -348,17 +348,97 @@ run "aws_db_instance_unit_test" {
...
+  }
+
+  assert {
+    condition     = length(module.database-load) == 0
+    error_message = "database-load module should be created"
```
Comment (ksugden) [inline postgres/tests/unit.tftest.hcl:440]:
```suggestion
    error_message = "database-load module should not be created"
```

File: postgres/tests/unit.tftest.hcl [hunk 6]
```diff
@@ -391,17 +471,56 @@ run "aws_db_instance_unit_test_database_load_created" {
     }
   }
 
-  # Uncomment the assertions below once DBTP-1427 is solved
+  assert {
+    condition     = length(module.database-dump) == 0
+    error_message = "database-dump module should be not created"
```
Comment (ksugden) [inline postgres/tests/unit.tftest.hcl:476]:
```suggestion
    error_message = "database-dump module should not be created"
```

File: postgres/tests/unit.tftest.hcl [hunk 7]
```diff
@@ -391,17 +471,56 @@ run "aws_db_instance_unit_test_database_load_created" {
...
+          from = "some-other-environment"
+          to   = "test-prod-environment"
+        }
+      ]
+    }
+  }
+
+  assert {
+    condition     = length(module.database-dump) == 0
+    error_message = "database-dump module should be not created"
```
Comment (ksugden) [inline postgres/tests/unit.tftest.hcl:517]:
```suggestion
    error_message = "database-dump module should not be created"
```

File: postgres/tests/unit.tftest.hcl [hunk 8]
```diff
@@ -391,17 +471,56 @@ run "aws_db_instance_unit_test_database_load_created" {
...
+  }
 
-  # assert {
-  #   condition     = length(module.database-load) == 1
-  #   error_message = "database-load module should be created"
-  # }
+  assert {
+    condition     = length(module.database-load) == 0
+    error_message = "database-load module should be created"
```
Comment (ksugden) [inline postgres/tests/unit.tftest.hcl:522]:
```suggestion
    error_message = "database-load module should not be created"
```

File: postgres/database-load/main.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (ksugden) [reply postgres/database-load/main.tf:75]: there should be a separate local for load and dump

**Pull request merged**